### PR TITLE
[luci/service] Migrate StridedSlice shape inference rule

### PR DESCRIPTION
**Pull request by qsunki**
This migrates StridedSlice shape inference rule to sinf::Algorithm.

ONE-DCO-1.0-Signed-off-by: sunki <ajsthfldu@outlook.com>

### CODE REVIEW

**Comment by qsunki** [comment]
for: #13932 
draft: #13914

File: compiler/luci/service/src/ShapeInfer_StridedSlice.cpp [hunk 1]
```diff
@@ -1,444 +0,0 @@
-/*
- * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
- * Copyright 2018 The TensorFlow Authors. All Rights Reserved.
```
Comment (seanshpark) [inline compiler/luci/service/src/ShapeInfer_StridedSlice.cpp:3]: plz copy this copyright from TensorFlow

File: compiler/luci/service/src/Nodes/CircleStridedSlice.cpp [hunk 2]
```diff
@@ -33,4 +46,413 @@ luci::CircleNode *CloneNodeLet<CN::STUV>::visit(const luci::CircleStridedSlice *
...
+loco::TensorShape Algorithm::visit(const luci::CircleStridedSlice *node)
+{
+  loco::TensorShape output_shape;
+
+  auto input_node = loco::must_cast<luci::CircleNode *>(node->input());
+
+  auto begin_node = dynamic_cast<luci::CircleConst *>(node->begin());
+  auto end_node = dynamic_cast<luci::CircleConst *>(node->end());
+  auto strides_node = dynamic_cast<luci::CircleConst *>(node->strides());
+  // TODO support non-const case
```
Comment (shs-park) [inline compiler/luci/service/src/Nodes/CircleStridedSlice.cpp]: Everything seems to be the same except this comment was added.
👍

**Comment by shs-park** [comment]
Please note that this PR also comes from issue https://github.com/Samsung/ONE/issues/13697.

**Review by shs-park — APPROVED** [review]

**Review by seanshpark — APPROVED** [review]
LGTM

File: compiler/luci/service/src/Nodes/CircleStridedSlice.cpp [hunk 3]
```diff
@@ -92,19 +93,75 @@ struct StridedSliceContext
...
+    assert(begin->size<S32>() <= input_dims);
+    assert(end->size<S32>() <= input_dims);
+    assert(strides->size<S32>() <= input_dims);
   }
   StridedSliceParams params;
   luci::CircleNode *input = nullptr;
   luci::CircleConst *begin = nullptr;
   luci::CircleConst *end = nullptr;
   luci::CircleConst *strides = nullptr;
+  luci::CircleConst dummy;
```
Comment (shs-park) [inline compiler/luci/service/src/Nodes/CircleStridedSlice.cpp:164]: @qsunki,
These changes are not related to the adding copyright part!?
Could it be that it was added incorrectly?

Comment (shs-park) [reply compiler/luci/service/src/Nodes/CircleStridedSlice.cpp:164]: This code was added while I was reviewing it, so I checked this part late. 😭

Comment (qsunki) [reply compiler/luci/service/src/Nodes/CircleStridedSlice.cpp:164]: Oh... another commit that I was testing got included... How should I fix this?

Comment (seanshpark) [reply compiler/luci/service/src/Nodes/CircleStridedSlice.cpp:164]: Post a PR to remove your unintended changes.

**Comment by shs-park** [comment]
@qsunki,
This is already merged PR.
Please create a new PR to restore your unintended code.
=)
